packages/at_onboarding_flutter/lib/screen/at_onboarding_generate_screen.dart

```diff
@@ -494,15 +495,19 @@ class _AtOnboardingGenerateScreenState
         if (!mounted) return;
         Navigator.pop(
             context, AtOnboardingResult.success(atsign: verifiedAtSign));
-      } else if (authResponse == AtOnboardingResponseStatus.serverNotReached) {
+      } else {
+        _onboardingService.setAtsign = previousAtsign;
+      }
+
+      if (authResponse == AtOnboardingResponseStatus.serverNotReached) {
         await _showAlertDialog(
           AtOnboardingLocalizations.current.msg_atSign_unreachable,
         );
       } else if (authResponse == AtOnboardingResponseStatus.authFailed) {
         await _showAlertDialog(
           AtOnboardingLocalizations.current.error_authenticated_failed,
         );
-      } else {
+      } else if (authResponse != AtOnboardingResponseStatus.authSuccess) {
```

Comment:
breaking the if-else chain requires an explicit check for non-success on our catch-all error check